src/odbc/src/app/application_data_buffer.cpp

```diff
@@ -278,15 +279,14 @@ ConversionResult::Type ApplicationDataBuffer::PutStrToStrBuffer(
   }
 
   SqlUlen currentCellOffset = cellOffset >= 0 ? cellOffset : 0;
-  // Since cellOffset is in bytes, an index needs to be calculated.
-  SqlUlen inCharIndex = currentCellOffset / inCharSize;
 
-  if (inCharIndex >= value.length()) {
-    if (resLenPtr) {
-      *resLenPtr = SQL_NO_TOTAL;
-    }
+  // If all data has already been read, return AI_NO_DATA
+  if (((currentCellOffset / inCharSize) * outCharSize) >= bytesRequired) {
     return ConversionResult::Type::AI_NO_DATA;
   }
+  
+  // Since cellOffset is in bytes, an index needs to be calculated.
+  SqlUlen inCharIndex = currentCellOffset / inCharSize;
```

Comment:
This gets calculated twice. Here and at line 284.
Also please expand the names (like specify that the index is in code units or encoding, if it's the source or target, etc)

Reply:
Updated. to not re-calculate the index, in fact it's a meaningless call anyways. inCharSize is always 1, so all we care about is the currentCellOffset.